pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -588,6 +592,9 @@ func (gb *GraphBuilder) processGraphChanges() bool {
 	glog.V(5).Infof("GraphBuilder process object: %s/%s, namespace %s, name %s, uid %s, event type %v", event.gvk.GroupVersion().String(), event.gvk.Kind, accessor.GetNamespace(), accessor.GetName(), string(accessor.GetUID()), event.eventType)
 	// Check if the node already exsits
 	existingNode, found := gb.uidToNode.Read(accessor.GetUID())
+	if found {
+		existingNode.markObserved()
```

Comment:
Ask discussed on slack, this is only safe because of callsite behavior.  The method and workqueue are private and its not unusual in this area of code, but it is comment worthy.  

This is what triggers marking the node observed in the "good" case when the informer sees the object.

Reply:
added comments, and tightened up call sites to only expose enqueuing virtual delete events into graphChanges outside the normal informer paths